soundata/core.py

```diff
@@ -361,6 +377,459 @@ def validate(self, verbose=True):
         )
         return missing_files, invalid_checksums
 
+    @lru_cache(maxsize=None)  # Setting maxsize to None for an unbounded cache
+    def compute_clip_statistics(self):
+        durations = [
+            len(self.clip(c_id).audio[0]) / self.clip(c_id).audio[1]
+            for c_id in tqdm(
+                list(self._index["clips"].keys())[:100], desc="Calculating durations"
+            )
+        ]
+
+        # Calculate statistics
+        total_duration = sum(durations)
+        mean_duration = np.mean(durations)
+        median_duration = np.median(durations)
```

Comment:
IIRC there was a problem with one of this that was showing the wrong estimate, is that working now?